File: examples/typescript/quickstart/workflows/first-task.ts
import { hatchet } from '../../hatchet-client';

type SimpleInput = {
  Message: string;
};

type SimpleOutput = {
  TransformedMessage: string;
};

export const firstTask = hatchet.task({
  name: 'first-task',
  fn: (input: SimpleInput, ctx): SimpleOutput => {
    return {
      TransformedMessage: input.Message.toLowerCase(),
    };
  },
});

File: examples/ruby/quickstart/workflows/first_task.rb
# frozen_string_literal: true

require "hatchet-sdk"

HATCHET = Hatchet::Client.new unless defined?(HATCHET)

# > Simple task
FIRST_TASK = HATCHET.task(name: "first-task") do |input, ctx|
  puts "first-task called"
  { "transformed_message" => input["message"].downcase }
end
